### PR TITLE
Autopick port & keep track of an app being served

### PR DESCRIPTION
**Pull request by sivakumar-kailasam**
[Rendered](https://github.com/sivakumar-kailasam/rfcs/blob/autopick-available-port/active/0000-autopick-available-port.md)

### CODE REVIEW

File: active/0000-autopick-available-port.md [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+Users might not pay attention to the build output by default when they run `ember serve` and might just prefer to see a failure to launch than us handling it.
+
+# Alternatives
+
+Users will have to figure out if another process is using the default port vs their ember app already running and proceed to work around it themselves. 
+
+# Unresolved questions
+
+In live reload's case incrementing by 1 doesn't matter as most users wont care which port live reload works out of. But if port `4200` is not available, what intervals do we use to pick the next available port? This might be important since we'd want the port number to be simple enough to remember.
```
Comment (stefanpenner) [inline active/0000-autopick-available-port.md]: I suspect if we auto-choose, we should be careful too not risk colliding with other less flexible systems (which may attempt to start up after this server starts). For that, selecting a port in the unassigned range 49152–65535 is most appropriate.

**Comment by sivakumar-kailasam** [comment]
@stefanpenner updated based on your feedback

**Comment by sivakumar-kailasam** [comment]
@stefanpenner if this looks good, I'll get to raising a PR

**Comment by nathanhammond** [comment]
We support this behavior in the most recent beta if you specify `ember s --port 0`. It will automatically choose the port for you and then launch the server. Does that meet your goal, @sivakumar-kailasam?

**Comment by sivakumar-kailasam** [comment]
@nathanhammond Wasn't aware of that. Noticed the relevant discussion in this PR, ember-cli/ember-cli/pull/5391. Is that behavior documented? Would it make sense for it to be added in the [port option's description](https://github.com/ember-cli/ember-cli/blob/master/lib/commands/serve.js#L22)?

**Comment by stefanpenner** [comment]
> is that behavior documented? Would it make sense for it to be added in the port option's description?

yes most likely

**Comment by sivakumar-kailasam** [comment]
Ok will close this in favor of a PR to add it to the description

**Pull request closed without merging**